### ISSUE TITLE
FR: Would it be possible to integrate with CrowdSec?

### ISSUE THREAD

**Issue by kameo4242** [tailscale/tailscale#7050]
### What are you trying to do?

I'd like to use Tailscale along with CrowdSec.
The crowd source blocklist is very efficient to block any BF attempt and if the IP is not yet known the agent can detect abnormal behaviors in logs and block them as well (not unlike fail2ban, but 2023 & on steroids).

### How should we solve this?

contact crowdsec team on https://discord.gg/crowdsec

### What is the impact of not solving this?

less security against IPs trying to BF or abnormal auth attempts

### Anything else?

nope, brilliant project

**Comment by bradfitz** [comment]
BF = brute force? (I associate BF with my initials :))

**Comment by kameo4242** [comment]
Ooops sorry yes. Bruteforce. 
Or anything nefarious security-wise that can be logged actually.

**Comment by DentonGentry** [comment]
The suggestion is to use CrowdSec on IP addresses of devices attempting to authenticate on the tailnet?

**Comment by kameo4242** [comment]
yes. The security engine could scan the logs to search for auth errors, L7 ddos and the likes and the real-time community blocklist would reject incoming IPs as long as they are marked as bad.

**Comment by DentonGentry** [comment]
There is an API that an external CrowdSec facility could use to authorize or deauthorize a device for the tailnet: https://github.com/tailscale/tailscale/blob/main/api.md#device-authorized-post
